Magento2/Sniffs/Legacy/AbstractBlockSniff.php

```diff
@@ -0,0 +1,97 @@
+<?php
+declare( strict_types = 1 );
+/**
+ * Copyright © Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento2\Sniffs\Legacy;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+class AbstractBlockSniff implements Sniff
+{
+    private const CHILD_HTML_METHOD = 'getChildHtml';
+    private const CHILD_CHILD_HTML_METHOD = 'getChildChildHtml';
+
+    /**
+     * Warning violation code.
+     *
+     * @var string
+     */
+    protected $errorCode = 'FoundCountOfParametersIncorrect';
+
+    /**
+     * @inheritdoc
+     */
+    public function register(): array
+    {
+        return [
+            T_OBJECT_OPERATOR
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        if (!isset($phpcsFile->getTokens()[$stackPtr + 1]['content'])) {
+            return;
+        }
+
+        $content = $phpcsFile->getTokens()[$stackPtr + 1]['content'];
+
+        if (!$this->isApplicable($content)) {
+            return;
+        }
+        
+        $paramsCount = $this->getParametersCount($phpcsFile, $stackPtr + 1);
+        if ($content === self::CHILD_HTML_METHOD && $paramsCount >= 3) {
+            $phpcsFile->addError(
+                '3rd parameter is not needed anymore for getChildHtml()',
+                $stackPtr,
+                $this->errorCode
+            );
+        }
+        if ($content === self::CHILD_CHILD_HTML_METHOD && $paramsCount >= 4) {
+            $phpcsFile->addError(
+                '4th parameter is not needed anymore for getChildChildHtml()',
+                $stackPtr,
+                $this->errorCode
+            );
+        }
+    }
+
+    /**
+     * Return if it is applicable to do the check
+     *
+     * @param String $content
```

Comment:
Not to mistake with class name
```suggestion
     * @param string $content
```

Reply:
Done